pmm/v3/pmm3-watchtower-autobuild.groovy

```diff
@@ -0,0 +1,122 @@
+library changelog: false, identifier: 'lib@master', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
+]) _
+
+pipeline {
+    agent {
+        label 'agent-amd64-ol9'
+    }
+    parameters {
+        string(
+            defaultValue: 'v3',
+            description: 'Tag/Branch for pmm-submodules repository',
+            name: 'GIT_BRANCH'
+        )
+        choice(
+            choices: ['experimental', 'testing'],
+            description: 'Select the destination environment for tagging the image.',
+            name: 'DESTINATION'
+        )
+    }
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '30'))
+        skipDefaultCheckout()
+        disableConcurrentBuilds()
+        parallelsAlwaysFailFast()
+    }
+    triggers {
+        upstream upstreamProjects: 'pmm3-submodules-rewind', threshold: hudson.model.Result.SUCCESS
+    }
+    environment {
+        PATH_TO_WATCHTOWER = 'sources/watchtower/src/github.com/percona/watchtower'
+    }
+    stages {
+        stage('Prepare') {
+            steps {
+                git poll: true, branch: GIT_BRANCH, url: 'http://github.com/Percona-Lab/pmm-submodules'
+                script {
+                    env.VERSION = sh(returnStdout: true, script: "cat VERSION").trim()
+                }
+                sh '''
+                    git reset --hard
+                    git clean -xdf
+                    git submodule update --init --jobs 10 ${PATH_TO_WATCHTOWER}
+                    git submodule status
+
+                '''
+                script {
+                    if (params.DESTINATION == "testing") {
+                        env.WATCHTOWER_LATEST_TAG   = "${VERSION}-rc${BUILD_NUMBER}"
+                        env.WATCHTOWER_RC_TAG       = "${VERSION}-rc"
+                    } else {
+                        env.WATCHTOWER_LATEST_TAG   = "dev-latest"
+                    }
+                }
+            }
+        }
+        stage('Build watchtower binary') {
+            steps {
+                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'pmm-staging-slave', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                    sh '''
+                        docker run --rm \
+                            -v ${WORKSPACE}/${PATH_TO_WATCHTOWER}:/watchtower \
+                            public.ecr.aws/e7j3v3n0/rpmbuild:3 \
+                            /bin/bash -c "
+                                cd /watchtower && \
+                                GO111MODULE=on CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-extldflags \\"-static\\" -X github.com/containrrr/watchtower/internal/meta.Version=$(git describe --tags || echo 'unknown')' .
+                            "
```

Comment:
To keep it clean here, I'd rather replace this whole command with a make target from that repo. Maybe need to create it first, if it doesn't exist.